File: library-client/src/main/java/org/internship/library/client/BookModel.java
package org.internship.library.client;

import org.internship.library.api.Book;

public class BookModel implements Book {

    private String id;
    private String title;
    private String author;
    private Integer numberOfPages;

    public BookModel() {
    }

    @Override
    public String getId() {
        return id;
    }

    @Override
    public void setId(String id) {
        this.id = id;
    }

    @Override
    public String getTitle() {
        return title;
    }

    @Override
    public void setTitle(String title) {
        this.title = title;
    }

    @Override
    public String getAuthor() {
        return author;
    }

    @Override
    public void setAuthor(String author) {
        this.author = author;
    }

    @Override
    public Integer getNumberOfPages() {
        return numberOfPages;
    }

    @Override
    public void setNumberOfPages(Integer numberOfPages) {
        this.numberOfPages = numberOfPages;
    }
}

File: library-frontend/src/app/model/book.model.ts

export class Book {
    id ?: string;
    author ?: string;
    title ?: string;
    numberOfPages ?: number;

    constructor(object: any) {
        this.id = object.id;
        this.author = object.author;
        this.title = object.title;
        this.numberOfPages = object.numberOfPages;
    }
}